Assets/Scripts/HealthManager.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class HealthManager : MonoBehaviour
+{
+    public static int health;
+    public Image[] hearts;
+
+    public Sprite fullHeart;
+    public Sprite emptyHeart;
+
+    private void Awake()
+    {
+        health = 3;
+    }
+
+    private void Update()
+    {
```

Comment:
Why are you calling this method in Update method? Just call this method whenever your player loses health.